### ISSUE TITLE
Using tagged templates results in no error reported

### ISSUE THREAD

**Issue by srg-kostyrko** [JSMonk/hegel#105]
As soon as tagged template is used hegel stops returning error information.

For example 

```
// Error: Type "'1'" is incompatible with type "number"
1 + '1';
```

With tagged template

```
// no errors at all
1 + '1';
a``;
```

Type inference seems to work, so the problem should be at error checking.

If I run CLI against file with such code I see only `TaggedTemplateExpression` in output.

**Comment by JSMonk** [comment]
Oh, okay. It seems like Hegel doesn't work with TaggedTemplateExpression. We will fix it soon.
Thank you for your contribution ^_^.

**Comment by JSMonk** [comment]
Fixed in `0.0.43` release. Thank you for the contribution ^_^.
